src/OrleansCodeGenerator/SerializerGenerator.cs

```diff
@@ -193,6 +193,22 @@ private static MemberDeclarationSyntax GenerateDeserializerMethod(Type type, Lis
                             SF.VariableDeclarator("boxedResult").WithInitializer(SF.EqualsValueClause(resultVariable)))));
                 boxedResultVariable = SF.IdentifierName("boxedResult");
             }
+            
```

Comment:
The crux of the change: generate code which calls `DeserializationContext.Current.RecordObject` after creating the object being deserialized. Note that we record `boxedResultVariable` in order to handle value types.